liboptv/src/correspondences.c

```diff
@@ -0,0 +1,709 @@
+                }
+            }
+        }
+    }
+    return matched;
+}
+
+int consistent_pair_matching(correspond *list[4][4], int num_cams, 
+    int *target_counts, double accept_corr, n_tupel *scratch, int scratch_size,
+    int** tusage)
+{
+    int matched = 0;
+    int i1, i2, n; /* camera indices */
+    int i, p1, p2;
+    double corr;
+    
+    for (i1 = 0; i1 < num_cams - 1; i1++) {
+        for (i2 = i1 + 1; i2 < num_cams; i2++) {
+            for (i=0; i < target_counts[i1]; i++) {
+                p1 = list[i1][i2][i].p1;
+                if (p1 > nmax || tusage[i1][p1] > 0) continue;
+
+                /* if the candidate is the only one, there is no ambiguity. 
+                   we only take unambiguous pairs.
+                */
+                if (list[i1][i2][i].n != 1) continue;
+
+                p2 = list[i1][i2][i].p2[0];
+                if (p2 > nmax || tusage[i2][p2] > 0) continue;
+
+                corr = list[i1][i2][i].corr[0] / list[i1][i2][i].dist[0];
+                if (corr <= accept_corr) continue;
+
+                /* This to catch the excluded cameras */
+                for (n = 0; n < num_cams; n++) 
+                    scratch[matched].p[n] = -2;
+                
+                scratch[matched].p[i1] = p1;
+                scratch[matched].p[i2] = p2;
+                scratch[matched].corr = corr;
+                
+                matched++;
+                if (matched == scratch_size) {
+                    printf ("Overflow in correspondences.\n");
+                    return matched;
+                }
+            }
+        }
+    }
+    return matched;
+}
+
+/****************************************************************************/
+/*         Other components of the correspondence process                   */
+/****************************************************************************/
+
+/*  match_pairs() finds for each target in each source camera the candidate
+    targets for correspondence on the other cameras by projecting epipolar 
+    lines.
+    
+    Arguments:
+    correspond *list[4][4] - pairwise adjacency lists to be filled with the
+        results. Each is a buffer long enough to contain data for all targets
+        in the source camera of the pair.
+    coord_2d **corrected - the flat-image metric coordinates of targets, by
+        camera, x-sorted. Epipolar lines are projected in flat coordinates.
+    frame *frm - frame data and reference target properties (similarity of
+        size and brightness are used to determine correlation score).
+    volume_par *vpar, control_par *cpar, Calibration **calib - scene 
+        parameters.
+*/
+void match_pairs(correspond *list[4][4], coord_2d **corrected, 
+    frame *frm, volume_par *vpar, control_par *cpar, Calibration **calib) 
+{
+    int i1, i2, i, j, pt1, count;
+    double xa12, ya12, xb12, yb12; /* Epipolar line edges */
+    candidate cand[MAXCAND];
+    
+    for (i1 = 0; i1 < cpar->num_cams - 1; i1++) {
```

Comment:
is it possible to do this bi-directional? wouldn't it improve substantially the quality of the correspondences?